source/slang/slang.cpp

```diff
@@ -372,12 +372,12 @@ SlangResult CompileRequest::loadFile(String const& path, ISlangBlob** outBlob)
 
 RefPtr<Expr> CompileRequest::parseTypeString(TranslationUnitRequest * translationUnit, String typeStr, RefPtr<Scope> scope)
 {
-    Slang::SourceFile srcFile;
-    srcFile.content = UnownedStringSlice(typeStr.begin(), typeStr.end());
+    Slang::RefPtr<Slang::SourceFile> srcFile = sourceManager->newSourceFile(String("type string"), typeStr);
```

Comment:
One unfortunate detail here is that the new source file we create when parsing a type name string for the purposes of the reflection API is getting added to the source manager in a long-term fashion, rather than being transient. That means that an application that compiles code and then queries a bunch of types by name could actually exhaust the available source ranges over many calls.

Ideally each type-name parse should somehow use a "sub" source manager that is transient for the duration of the request and just inherits the existing used source ranges and their meanings from the parent.